mysql-test/lib/generate_percona_ssl_certs.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+set -xe
+
+# boilerplace for "openssl ca" and /etc/ssl/openssl.cnf
+rm -rf demoCA
+mkdir demoCA demoCA/newcerts
+touch demoCA/index.txt
+echo 01 > demoCA/serial
+
+# Generate a new self-signed CA certificate
+openssl req -x509 -newkey rsa:2048 -keyout percona-cakey.pem -out percona-cacert.pem -days 7300 -nodes -subj '/CN=percona-cacert/C=US/ST=NC/L=Raleigh/O=Percona' -text
+# The following operation, althought looking redundant, will remove 'passphrase' from the RSA key (which is currently unsupported in YaSSL)
+openssl rsa -in percona-cakey.pem -out percona-cakey.pem
+
+# Generating a new certificate signing request for the server
+openssl req -newkey rsa:2048 -keyout percona-serversan-key.pem -out demoCA/percona-serversan-req.pem -days 7300 -nodes -subj '/CN=percona-server/C=US/ST=NC/L=Raleigh/O=Percona'
+# The following operation, althought looking redundant, will remove 'passphrase' from the RSA key (which is currently unsupported in YaSSL)
+openssl rsa -in percona-serversan-key.pem -out percona-serversan-key.pem
+# Generate a new server certificate signed with new CA
+# with SubjectAltName, only for OpenSSL 1.0.2+
+cat > demoCA/sanext.conf <<EOF
+subjectAltName=DNS:localhost
```

Comment:
1. Apart from DNS:localhost subjectAltName can also take IP addresses. Please add test for IP also, for instance IP:127.0.0.1.
2. Also we should have a test certificate which would allow multiple subjectAltNames and it should be tested